cmd/xcode.go

```diff
@@ -108,6 +114,67 @@ func scanXcodeProject(_ *cobra.Command, _ []string) error {
 		xcodeCmd.SDK = paramXcodebuildSDK
 	}
 
+	if paramXcodeDestination != "" {
+		xcodeCmd.DESTINATION = paramXcodeDestination
+	} else {
+		var project xcodeproj.XcodeProj
+		var scheme xcscheme.Scheme
+
+		if xcodeproj.IsXcodeProj(xcodeCmd.ProjectFilePath) {
```

Comment:
we already have a working logic for opening a project, scheme, configuration, based on the provided parameters: https://github.com/bitrise-steplib/steps-xcode-archive/blob/master/utils/platform.go#L25-L53

Reply:
and for reading the archivable target's platform: https://github.com/bitrise-steplib/steps-xcode-archive/blob/master/utils/platform.go#L70-L92

Reply:
updated some parts and will create separate ticket